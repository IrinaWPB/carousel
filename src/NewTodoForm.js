import React, { useState } from 'react'
import {v4 as uuid} from 'uuid'

const NewTodoForm = ({ addTodo }) => {
    const [todo, setTodo] = useState("")

    const handleChange = e => {
        setTodo(e.target.value)
    }
    
    const handleSubmit = e => {
        e.preventDefault();
        addTodo({ todo, id: uuid() });
        setTodo("")
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="todo">Add a new task:</label>
            <input id="todo" type="text" name="todo" value={todo} onChange={handleChange} />
            <button>Add Task</button>
        </form>
    )
}

export default NewTodoForm