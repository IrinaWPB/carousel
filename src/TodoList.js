import React, { useState } from 'react'
import Todo from './Todo'
import NewTodoForm from './NewTodoForm'


const TodoList = () => {
    const [todos, setTodos] = useState([])

    //function to run when new data is submited, adds a new todo to the list.
    const addTodo = newTodo => {
        setTodos(todos => [...todos, newTodo])
    }
    
    const removeTodo = id => {
        setTodos(todos => todos.filter(todo => todo.id !== id))
    }

    const updateTodo = (id, updatedTask) => {
        setTodos(todos => todos.map(todo => todo.id === id ? {...todo, todo: updatedTask } : todo ))
    }

    return (
        <div className='TodoList'>
            <h1>My Todo List</h1>
            <div className='TodoList-form-div'>
                <NewTodoForm addTodo={addTodo} />
            </div>
            <div className='TodoList-ul'>
                <ul data-testid="list">
                    {todos.map(todo => <Todo id={todo.id} 
                                             text={todo.todo} 
                                             key={todo.id} 
                                             removeTask={removeTodo} 
                                             updateTask={updateTodo} />)}
                </ul>    
            </div>       
        </div>
    )
}

export default TodoList