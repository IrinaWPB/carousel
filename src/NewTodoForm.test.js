import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import NewTodoForm from './NewTodoForm'
import TodoList from './TodoList'

it('renders without crashing', () => {
    render(<NewTodoForm />)
})

it('matches the snapshot', () => {
    const {asFragment} = render(<NewTodoForm />)
    expect(asFragment()).toMatchSnapshot()
})
