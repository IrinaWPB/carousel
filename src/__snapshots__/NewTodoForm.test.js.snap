// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="todo"
    >
      Add a new task:
    </label>
    <input
      id="todo"
      name="todo"
      type="text"
      value=""
    />
    <button>
      Add Task
    </button>
  </form>
</DocumentFragment>
`;
